tests/test_integration_api2_fairir.py

```diff
@@ -0,0 +1,1258 @@
+        content_type="application/json",
+        headers=openreview_client.headers,
+    )
+    assert response.status_code == 200
+
+    matcher_status = wait_for_status(
+        openreview_client, config_note["note"]["id"], api_version=2
+    )
+    assert matcher_status.content["status"]["value"] == "Complete"
+
+    paper_assignment_edges = openreview_client.get_edges_count(
+        label="integration-test",
+        invitation=venue.get_assignment_id(venue.get_reviewers_id()),
+    )
+
+    assert paper_assignment_edges == num_papers * reviews_per_paper
+
+def test_integration_attribute_constraints(
+    openreview_context, celery_app, celery_worker
+):
+    """
+    Basic integration test. Makes use of the OpenReview Builder
+    """
+    openreview_client = openreview_context["openreview_client_v2"]
+    test_client = openreview_context["test_client"]
+
+    conference_id = "IMCLR.ws/2017/Conference"
+    num_reviewers = 3
+    num_papers = 1
+    reviews_per_paper = 3
+    max_papers = 1
+    min_papers = 1
+    alternates = 0
+
+    venue = clean_start_conference_v2(
+        openreview_client,
+        conference_id,
+        num_reviewers,
+        num_papers,
+        reviews_per_paper,
+    )
+
+    post_fairir_to_api2(openreview_client, conference_id)
+
+    reviewers_id = venue.get_reviewers_id()
+    seniority_inv_id = "{}/-/Seniority".format(reviewers_id)
+
+    config = {
+        "title": {"value": "integration-test-constraints"},
+        "user_demand": {"value": str(reviews_per_paper)},
+        "max_papers": {"value": str(max_papers)},
+        "min_papers": {"value": str(min_papers)},
+        "alternates": {"value": str(alternates)},
+        "config_invitation": {
+            "value": "{}/-/Assignment_Configuration".format(reviewers_id)
+        },
+        "paper_invitation": {"value": venue.get_submission_id()},
+        "assignment_invitation": {
+            "value": venue.get_assignment_id(reviewers_id)
+        },
+        "deployed_assignment_invitation": {
+            "value": venue.get_assignment_id(reviewers_id, deployed=True)
+        },
+        "invite_assignment_invitation": {
+            "value": venue.get_assignment_id(reviewers_id, invite=True)
+        },
+        "aggregate_score_invitation": {
+            "value": "{}/-/Aggregate_Score".format(reviewers_id)
+        },
+        "conflicts_invitation": {
+            "value": venue.get_conflict_score_id(reviewers_id)
+        },
+        "custom_max_papers_invitation": {
+            "value": "{}/-/Custom_Max_Papers".format(reviewers_id)
+        },
+        "match_group": {"value": reviewers_id},
+        "scores_specification": {
+            "value": {
+                venue.get_affinity_score_id(reviewers_id): {
+                    "weight": 1.0,
+                    "default": 0.0,
+                }
+            }
+        },
+        "constraints_specification": {
+            "value": {
```

Comment:
this constraint seems to have another schema that the one defined in the PR, am I missing something?

```
{
  "constraintName": {
    "bound": int,
    "comparator": str,
    "members": List[str]
  }
}

```

Reply:
Sorry for the confusion, I updated the PR with the edge based constraints we discussed but to explain more:

1) The matcher reads the edges from the API using the `constraints_specification` into this form:
```
{
  "constraintName": {
    "bound": int,
    "comparator": str,
    "members": List[str]
  }
}
```
and each constraint name is based on the edge invitation and the labels provided in `constraints_specification` - so from your previous comments, one of them would be "Seniority/junior"

2) Then the encoder turns it into a form that's understood by the Gurobi matcher:
```
[
  {
    "name": "constraintName",
    "bound": int,
    "comparator": str,
    "members": List[int] // Maps reviewer IDs to an index stored in the encoder
  }
]
```

Reply:
got it!